### PR TITLE
PTK colors refactor

### LINKED ISSUE #3047: Enable new color modifiers on PTK
See https://github.com/xonsh/xonsh/pull/3045#issuecomment-471631973

### PR DESCRIPTION
**Pull request by scopatz**
Fixes  #3047

### CODE REVIEW

File: xonsh/pyghooks.py [hunk 1]
```diff
@@ -278,6 +280,63 @@ def color_by_name(name, fg=None, bg=None):
     return tok, fg, bg
 
 
+@lazyobject
+def PYGMENTS_MODIFIERS():
+    # pygments doesn't support all modifiers.
+    # use None to represent unsupported
```
Comment (melund) [inline xonsh/pyghooks.py]: I know pygments doesn't  support all the modifiers but prompt toolkit support `reverse` and `blink`. So maybe there was a way to use them still.

File: xonsh/pyghooks.py [hunk 2]
```diff
@@ -293,34 +352,22 @@ def code_by_name(name, styles):
     code : str
         Pygments style color code.
     """
-    fg, _, bg = name.lower().partition("__")
-    if fg.startswith("background_"):
+    fg, _, bg = name.upper().replace("HEX", "#").partition("__")
```
[the remark this answers is not in the document]
Comment (scopatz) [reply xonsh/pyghooks.py]: Yeah, I don't know why this was implemented in this way. For this PR, I just tried to match what was there already.

File: xonsh/pyghooks.py (hunk 1 above)
Comment (scopatz) [reply xonsh/pyghooks.py]: Yeah, I think we might want to have a separate module for computing PTK 2 colors directly from color names.

**Comment by scopatz** [comment]
Thanks for the review and merge @melund!

**Pull request merged**